lib/elixir/src/elixir_env.erl

```diff
@@ -107,9 +107,18 @@ merge_and_check_unused_vars(Current, Unused, ClauseUnused, E) ->
       end;
 
     ({{Name, Kind}, _Count}, {Meta, Overridden}, Acc) ->
-      case (Kind == nil) andalso is_unused_var(Name) of
+      DisplayName = case Kind of
+        nil ->
+          Name;
+        elixir_fn ->
+          <<"x", Rest>> = atom_to_binary(Name),
+          binary_to_atom(<<"&", Rest>>);
```

Comment:
Was the approach mentioned in the issue to just emit vars with `Kind = nil` in this case?
This could work too but would generate warnings for `x1` which might confuse users (although probably an improvement over today and people would guess I suppose).
By keeping `elixir_fn` we might be able to handle it better here.

Reply:
What if we say the name of the variable is literally :"&1" and keep kind as nil? Or would that print incorrectly?

Reply:
Seems to work fine! Trying to think of a potential downside but I can't either.